exploration/code-mode-introducer.md

```diff
@@ -0,0 +1,224 @@
+set (with syntax adjustments if we use up a private-use one).
+Most of the options below have been changed to use `^`, using 
+Apple's experimental syntax as a model for sigil choice.
+
+It should be noted that an introducer sigil should be as rare as possible in normal text.
+This tends to run against common punctuation marks `&`, `%`, `!`, and `?`.
+
+```abnf
+reserved-start = "!" / "@" / "#" / "%" / "*" / "<" / ">" / "/" / "?" / "~"
+private-start  = "^" / "&"
+```
+
+## Proposed Design
+
+_Describe the proposed solution. Consider syntax, formatting, errors, registry, tooling, interchange._
+
+We need to choose one of these (or another option not yet considered).
+Presentation at UTW did not produce any opinions.
+
+Based on the pro/cons below, I would suggest Option D is possibly the best option?
+
+## Alternatives Considered
+
+_What other solutions are available?_
+_How do they compare against the requirements?_
+_What other properties they have?_
+
+There are the following designs being considered:
+
+### Option A. Use Pattern Quotes for Messages (current design)
+
+Complex messages are quoted with double curly brackets.
+The closing curly brackets might be optional.
+
+Sample pattern:
+```
+{{
+input {$var}
+match {$var}
+when * {{Pattern}}
+}}
+```
+Sample quoted pattern with no declarations or match:
+```
+{{{{Pattern}}}}
+```
+
+Pros:
+- Uses a sigil `{` already present in the syntax
+- No additional escapes
+- Consistent with other parts of the syntax?
+
+Cons:
+- Somewhat verbose
+- Closing portion of the syntax adds no value;
+  could be a source of unintentional syntax errors
+- Messages commonly end with four `}}}}`
+
+### Option B. Use a Sigil
+
+Complex messages start with a special sigil character.
+
+```
+^input {$var}
+match {$var}
+when * {{Pattern}}
+```
+Sample quoted pattern with no declarations or match:
+```
+^{{Pattern}}
+```
+
+Pros:
+- Requires minimum additional typing
+
+Cons:
+- Requires an additional sigil
+- Requires an additional escape for simple pattern start
+- Has no other purpose in the syntax
+
+### Option C. Use a Double Sigil
+
+Like Option B, except the sigil is doubled.
+
+```
+^^input {$var}
+match {$var}
+when * {{Pattern}}
+```
+Sample quoted pattern with no declarations or match:
+```
+^^{{Pattern}}
+```
+
+Pros:
+- Less likely to conflict with a simple pattern
+
+Cons:
+- Requires an additional sigil
+- Requires an additional escape for simple pattern start
+- Has no other purpose in the syntax
+
+### Option D. Sigilized Keywords
+
+Instead of quoting the message, adds a sigil to keywords that
+start statements, that is, `#input`, `#local` and `#match`.
+The keyword `when` might be considered separately.
+
+```
+#input {$var}
+#local $foo = {$bar}
+#match {$var}
+when * {{Pattern}}
+```
+Sample quoted pattern with no declarations or match:
+```
+{{Pattern}}
+```
+
+Pros:
+- Sigil is part of the keyword, not something separate
+- Requires minimum additional typing
```

Comment:
I wouldn't say "minimum" here. Option B requires the absolute minimum additional typing, since it's just adding one character per message.

Reply:
Actually, Option D adds zero characters. That's actually the minimum ;-).

I could say "Requires **no** additional typing". Also, note that this option avoids some of the iteration hazard of other options.